### PR TITLE
[FLINK-35201][table] Support the execution of drop materialized table in full refresh mode

### PR DESCRIPTION
**Pull request by hackergin**
## What is the purpose of the change

*Support the execution of drop materialized table in full refresh mode*

## Brief change log

* Support the execution of drop materialized table in full refresh mode

## Verifying this change

* Add test case `testDropMaterializedTableInFullMode` to verify the execution of drop materialized table in full mode. 

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (Will be added in a separate pr)

### CODE REVIEW

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java [hunk 1]
```diff
@@ -782,35 +783,12 @@ private ResultFetcher callDropMaterializedTableOperation(
...
+                cancelContinuousRefreshJob(
+                        operationExecutor, handle, tableIdentifier, materializedTable);
             } else {
-                LOG.info(
-                        "No need to cancel the continuous refresh job {} for materialized table {} as it is not currently running.",
-                        refreshHandler.getJobId(),
-                        tableIdentifier);
+                deleteRefreshWorkflow(tableIdentifier, materializedTable);
             }
         } else if (CatalogMaterializedTable.RefreshStatus.INITIALIZING
```
Comment (lsyldliu) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/materializedtable/MaterializedTableManager.java:793]: If the refresh status is `SUSPENDED`, we also need to drop the refresh workflow before drop the materialized table.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/workflow/scheduler/EmbeddedQuartzScheduler.java [hunk 2]
```diff
@@ -271,11 +271,6 @@ public void deleteScheduleWorkflow(String workflowName, String workflowGroup)
         JobKey jobKey = JobKey.jobKey(workflowName, workflowGroup);
         lock.writeLock().lock();
         try {
-            String errorMsg =
-                    String.format(
-                            "Failed to delete a non-existent quartz schedule job: %s.", jobKey);
-            checkJobExists(jobKey, errorMsg);
-
             quartzScheduler.deleteJob(jobKey);
```
Comment (lsyldliu) [inline flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/workflow/scheduler/EmbeddedQuartzScheduler.java:274]: Do you verify whether  `quartzScheduler.deleteJob` throws an exception when deleting a not exist job?

**Review by lsyldliu — COMMENTED** [review]
@hackergin Thanks for contribution, LGTM overall, just left two minor comments.

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/workflow/scheduler/EmbeddedQuartzScheduler.java (hunk 2 above)
Comment (hackergin) [reply flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/workflow/scheduler/EmbeddedQuartzScheduler.java:274]: Yes, In the testDropMaterializedTableWithDeletedRefreshWorkflowInFullMode test, even if the workflow does not exist, `quartzScheduler.deleteJob(jobKey)` will return normally without error.

**Review by lsyldliu — APPROVED** [review]
LGTM

**Pull request merged**